packages/GA4Client/src/common.js

```diff
@@ -64,6 +71,48 @@ Common.prototype.truncateEventAttributes = function (eventAttributes) {
     );
 };
 
+Common.prototype.standardizeParameters = function (parameters) {
+    var standardizedParameters = {};
+    for (var key in parameters) {
+        var standardizedKey = this.standardizeName(key);
+        standardizedParameters[standardizedKey] = parameters[key];
+    }
+
+    return this.truncateAttributes(
+        standardizedParameters,
+        EVENT_ATTRIBUTE_KEY_MAX_LENGTH,
+        EVENT_ATTRIBUTE_VAL_MAX_LENGTH
+    );
+};
+
+Common.prototype.standardizeName = function (name) {
+    function removeNonAlphanumericCharacterFromStart(name) {
```

Comment:
Added the while loop

Reply:
it actually complicates things a bit but provides coverage for all use cases. I tried to simplify it as much as I could. I put functions inside since only this method uses them. I didn't think it made sense to put them on the `common` module, but it does complicate the method more.